docs: update monorepo guide for SDK 50 (#25848)
expo/expo: docs/pages/guides/monorepos.mdx

  path.resolve(projectRoot, 'node_modules'),
  path.resolve(workspaceRoot, 'node_modules'),
];

> /* @info Starting from Expo SDK 50, you no longer should disable the hierarchical lookup by default. Learn more in the section below. */

Reviewer: I think this should part of the main point above rather than having a highlight here in the code snippet (it's easier to miss this). We can also link the right section from the main context.

Author: Ok, so the reason why I put it in here, is for two things:

1. We don't emphasize anymore on "disable hierarchical lookup", so I'm not sure how well it reads when you start the main point with "Do NOT do this" (as opposed to "do this").
2. Since we _previously_ did recommend this, I kept it in the same spot, but disabled with more context on _why_ it's disabled.

If you think that's not good, I'm open for suggestions 😄

<Collapsible summary="3. Why do we need to disable the hierarchical lookup?">

<Callout type="warning">

>   Starting from Expo SDK 50, the hierarchical lookup should no longer be disabled by default. Both React Native and Expo has improved general monorepo support, and disabling this option can cause other issues. Only enable this option if you know what you are doing.

Reviewer: ```suggestion
  Starting from Expo SDK 50, the hierarchical lookup should no longer be disabled by default. Both React Native and Expo have improved general monorepo support, and disabling this option can cause other issues. Only enable this option if you know what you are doing.
```

It sounds a bit conflicting. So, if I am using SDK 50, I should no longer disable it. However, the last line says only enable it when I know what I'm doing. Can we either separate that and elaborate on it more?

Author: Yeah, but it's "should no longer be disabled by default". If you have reasons to, you can enable it. The "Only enable this option if you know what you are doing." could be removed I guess

Author: ```suggestion
  Starting from Expo SDK 50, the hierarchical lookup should no longer be disabled by default. Both React Native and Expo have improved general monorepo support, and disabling this option can cause other issues.
```

As mentioned earlier, using monorepos is not for everyone. You take on increased complexity and need to solve issues you most likely will run into. Here are a couple of common issues you might encounter.

### Multiple React Native versions within the monorepo

> Expo SDK 50 and higher has improved support for more complext `node_modules` patterns, such as isolated modules. Unfortunately, React Native itself can still cause issues when installing multiple versions inside a single monorepo. Because of that, it's recommended to only use a single version of React Native.

Reviewer: ```suggestion
Expo SDK 50 and higher has improved support for more complete **node_modules** patterns, such as isolated modules. Unfortunately, React Native can still cause issues when installing multiple versions inside a single monorepo. Because of that, it's recommended to only use a single version of React Native.
```

> You can check if your monorepo has multiple React Native versions, and why they are installed, through the package managers.

Reviewer: ```suggestion
You can check if your monorepo has multiple React Native versions and why they are installed through the package manager you use.
```

> <Terminal cmd={[

Reviewer: I also think we should use Tabs to separate diff pkg manage commands here so that folks can still copy paste it (just a thought, but should block this PR).

Author: That would be great, especially here!

Reviewer: I can follow up in another PR too as it should not be blocking this PR (sorry for the wrong context in my previous comment).

Author: No worries, let's do that!
